packages/pwa-kit-dev/src/configs/webpack/config.js

```diff
@@ -354,7 +354,11 @@ const renderer =
                 devtool: mode === development ? 'eval-source-map' : false,
                 output: {
                     path: buildDir,
-                    filename: 'server-renderer.js',
+
+                    // We want to split the build on local development to reduce memory usage.
+                    // It is required to have a single entry point for the remote server.
+                    // See pwa-kit-runtime/ssr/server/build-remote-server.js render method.
+                    filename: mode === development ? '[name].js' : 'server-renderer.js',
```

Comment:
🥳 nice! Wild to think that the magic of `[name].js` is the fix here. Can you unpack and explain that?

Reply:
@bfeister previously, our webpack build a single file `server-renderer.js` for SSR and this file is huge because it includes all dependencies and that file is about 50MB. Each time a change happens, webpack re-build this 50MB file and saves it in memory.

While we can't completely eliminate the memory leaks because it happens in 3rd party modules like `global._`, `global['__core-js-shared__']`, and there is also the bug in Node internal that actually holds on to references for dynamic imports. It's nearly impossible to eliminate the issue, but we can mitigate the issue by splitting the build into very small chunks, so each build would be much smaller, thus take less space in memory.

Reply:
This is amazing! I tested this PR on my local, not running into this issue once. Great job on finding the solution